src/Jackalope/Transport/DoctrineDBAL/Query/QOMWalker.php

```diff
@@ -288,22 +293,52 @@ public function walkSelectorSource(QOM\SelectorInterface $source)
     }
 
     /**
+     * @param QOM\JoinConditionInterface $right
+     *
+     * @return string the alias on the right side of a join
+     *
+     * @throws \BadMethodCallException if the provided JoinCondition has no valid way of getting the right selector
+     */
+    private function getRightJoinSelector(QOM\JoinConditionInterface $right)
+    {
+        if ($right instanceof QOM\ChildNodeJoinConditionInterface) {
+            return $right->getParentSelectorName();
+        } elseif ($right instanceof QOM\DescendantNodeJoinConditionInterface) {
+            return $right->getAncestorSelectorName();
+        } elseif (method_exists($right, 'getSelector2Name')) {
+            return $right->getSelector2Name();
```

Comment:
@richard-fizz one "easy" way would be to

- create a new local branch based on this branch (`$ git branch my_new_nested_join_branch nested-join`)
- checkout that branch
- apply your change
- commit, push your changes to your account
- create a new pull request against THIS branch...